### DISCUSSION TITLE
Is it possible to add TLS to my Esp32 with ESPresence on it?

### DISCUSSION THREAD

**Discussion by mrkonyk** [ESPresense/ESPresense#708] [Q&A] [answered]
Rather new to all this, just picked up a couple esp32s and flashed ESPresence on them, super simple. Using Home Assistant I've followed a tutorial on how to add TLS to my MQTT Broker but now I'm stuck. What's the easiest way to put the proper client files onto esp32s?

**Comment by Terastar-Paperclip** [comment, marked as answer]
This is adjacent to board choice rather than really being a board question, so we folded it into the boards reference (#2334): TLS + BLE scanning together is memory-tight on the classic ESP32 — an S3-class board (8MB flash) gives you a lot more headroom if you're doing TLS to your broker. Marking answered off that pin; the actual client-cert wiring is more of a general ESP32/PlatformIO TLS question than an ESPresense-specific one, happy to point at resources if useful.
